src/sentry/api/endpoints/organization_transaction_details.py

```diff
@@ -0,0 +1,363 @@
+from collections import defaultdict
+from datetime import datetime, timedelta, timezone
+
+import sentry_sdk
+from rest_framework.request import Request
+from rest_framework.response import Response
+from sentry_relay.consts import SPAN_STATUS_CODE_TO_NAME
+from snuba_sdk import Column, Condition, Function, Op
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.organization import OrganizationEndpoint
+from sentry.api.utils import handle_query_errors
+from sentry.constants import ObjectStatus
+from sentry.models.project import Project
+from sentry.search.events.builder.spans_indexed import SpansIndexedQueryBuilder
+from sentry.search.events.builder.spans_metrics import SpansMetricsQueryBuilder
+from sentry.search.events.types import ParamsType, QueryBuilderConfig
+from sentry.snuba.dataset import Dataset
+from sentry.snuba.referrer import Referrer
+
+VALID_AVERAGE_COLUMNS = {"span.self_time", "span.duration"}
+
+
+@region_silo_endpoint
+class OrganizationTransactionDetailsEndpoint(OrganizationEndpoint):
+    owner = ApiOwner.PERFORMANCE
+    publish_status = {
+        "GET": ApiPublishStatus.PRIVATE,
+    }
+
+    def get(self, request: Request, organization, project_slug, transaction_id):
+        try:
+            project = Project.objects.get(
+                slug=project_slug, organization_id=organization.id, status=ObjectStatus.ACTIVE
+            )
+        except Project.DoesNotExist:
+            return Response(status=404)
+
+        # Check access to the project as this endpoint doesn't use membership checks done
+        # get_filter_params().
+        if not request.access.has_project_access(project):
+            return Response(status=404)
+
+        # Get a list of all spans with this transaction_id
+        given_start_str = request.GET.get("start_timestamp", "")
+        given_end_str = request.GET.get("end_timestamp", "")
+        try:
+            start = datetime.fromtimestamp(float(given_start_str))
+            end = datetime.fromtimestamp(float(given_end_str))
+        except TypeError:
+            return Response({"detail": "missing start_timestamp or end_timestamp"}, status=400)
+
+        spans_data = _query_all_spans_in_transaction(
+            organization, project, transaction_id, start, end
+        )
+        if len(spans_data) == 0:
+            return Response(status=404)
+
+        segment_span = next((span for span in spans_data if span["is_segment"] != 0), None)
```

Comment:
Just to be safe, could we throw a warning here if there's more than one segment span?